app/controllers/registration_commands_controller.rb

```diff
@@ -44,6 +44,11 @@ def registration_params
     end
   end
 
+  def host_groups
+    Hostgroup.authorized(:view_hostgroups)
+             .each { |hg| hg.operatingsystem_id ||= hg.inherited_operatingsystem_id }
```

Comment:
I would expect that something like this would be solved in the model layer, not in a controller. However, it's really all over the place.

I think that in particular your solution to modify the Hostgroup object has a risk. If you save it, there's a risk of making the OS concrete instead of inherited. While today you may not do so, I think it's better to avoid that risk altogether.

I'm now looking at other reviewers/maintainers to weigh in this.

Reply:
I agree there should be a method on the model that would give me (self.os || self.inherited), such method should be used in the JS component instead of manually set operatingsystem_id. However the authorization should remain here.